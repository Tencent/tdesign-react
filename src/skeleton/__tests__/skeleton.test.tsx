// import { render } from '@test/utils';
// import React from 'react';
// import Skeleton from '../index';

// TODO
describe('Skeleton 组件测试', () => {
  test('dom', () => {
    expect(true).toBe(true);
  });
});
