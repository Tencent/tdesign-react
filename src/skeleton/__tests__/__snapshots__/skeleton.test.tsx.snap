// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`advance.jsx 1`] = `
<DocumentFragment>
  <div
    style="width: 100%;"
  >
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        网页（使用 rowCol）
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 33%; height: 180px;"
            >
              image
            </div>
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 33%; height: 180px;"
            >
              image
            </div>
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 33%; height: 180px;"
            >
              image
            </div>
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-circle"
              style="width: 50px; height: 50px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: calc(100% - 170px); height: 30px; margin: 20px 0px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 80px; height: 30px; margin-left: 20px;"
            >
              确定
            </div>
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="width: 70%; margin: 0px 0px 16px 0px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 100%; height: 300px;"
            >
              <div
                style="width: 100%; text-align: center;"
              >
                <h3
                  style="padding: 10px; border-bottom: 1px solid #ccc;"
                >
                  标题
                </h3>
                <div
                  style="padding: 10px; height: 200px;"
                >
                  内容
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        列表（组合使用）
      </div>
      <div
        style="padding: 16px;"
      >
        <ul>
          <li
            style="padding: 16px 0px;"
          >
            <div
              style="display: inline-block; margin-right: 20px;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-circle"
                  style="width: 56px; height: 56px;"
                />
              </div>
            </div>
            <div
              style="display: inline-block; width: 80%;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                  style="width: 70%;"
                />
              </div>
            </div>
          </li>
          <li
            style="padding: 16px 0px;"
          >
            <div
              style="display: inline-block; margin-right: 20px;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-circle"
                  style="width: 56px; height: 56px;"
                />
              </div>
            </div>
            <div
              style="display: inline-block; width: 80%;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                  style="width: 70%;"
                />
              </div>
            </div>
          </li>
          <li
            style="padding: 16px 0px;"
          >
            <div
              style="display: inline-block; margin-right: 20px;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-circle"
                  style="width: 56px; height: 56px;"
                />
              </div>
            </div>
            <div
              style="display: inline-block; width: 80%;"
            >
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                />
              </div>
              <div
                class="t-skeleton__row"
              >
                <div
                  class="t-skeleton__col t-skeleton--type-text"
                  style="width: 70%;"
                />
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`animation.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-skeleton-demo"
  >
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        渐变加载动画
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-gradient"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-gradient"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-gradient"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-gradient"
              style="width: 70%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        闪烁加载动画
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-flashed"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-flashed"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-flashed"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text t-skeleton--animation-flashed"
              style="width: 70%;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div>
    <div>
      <button
        class="t-switch t-is-checked"
        role="switch"
        style="margin-bottom: 20px;"
        type="button"
      >
        <span
          class="t-switch__handle"
        />
        <div
          class="t-switch__content"
        />
      </button>
    </div>
    <div>
      <div
        style="margin-bottom: 20px;"
      >
        <div
          class="t-skeleton__row"
        >
          <div
            class="t-skeleton__col t-skeleton--type-text"
          />
        </div>
        <div
          class="t-skeleton__row"
        >
          <div
            class="t-skeleton__col t-skeleton--type-text"
          />
        </div>
        <div
          class="t-skeleton__row"
        >
          <div
            class="t-skeleton__col t-skeleton--type-text"
          />
        </div>
        <div
          class="t-skeleton__row"
        >
          <div
            class="t-skeleton__col t-skeleton--type-text"
            style="width: 70%;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`theme.jsx 1`] = `
<DocumentFragment>
  <div
    class="t-skeleton-demo"
  >
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        text
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        avatar
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-circle"
              style="width: 56px; height: 56px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        paragraph
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="width: 70%;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        avatar-text
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-circle"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 32px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        tab
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 200px;"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 16px; border: 1px solid #eee;"
    >
      <div
        style="padding: 16px; border-bottom: 1px solid #eee;"
      >
        article
      </div>
      <div
        style="padding: 16px;"
      >
        <div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 100%; height: 30px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-rect"
              style="width: 100%; height: 200px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
          </div>
          <div
            class="t-skeleton__row"
          >
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
            <div
              class="t-skeleton__col t-skeleton--type-text"
              style="height: 30px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
